sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/LogDivertAppender.java

```diff
@@ -276,12 +276,19 @@ private static StringLayout initLayout(OperationLog.LoggingLevel loggingMode) {
     return getLayout(isVerbose, layout);
   }
 
-  public LogDivertAppender(OperationManager operationManager,
-    OperationLog.LoggingLevel loggingMode) {
+  public static LogDivertAppender create(OperationManager operationManager,
+                                         OperationLog.LoggingLevel loggingMode) {
+    CharArrayWriter writer = new CharArrayWriter();
+    return new LogDivertAppender(operationManager, loggingMode, writer);
+  }
+
+  private LogDivertAppender(OperationManager operationManager,
+    OperationLog.LoggingLevel loggingMode, CharArrayWriter writer) {
     super("LogDivertAppender", initLayout(loggingMode), null, false, true, Property.EMPTY_ARRAY,
-            new WriterManager(new CharArrayWriter(), "LogDivertAppender",
```

Comment:
The issue is here using another `CharArrayWriter`?

Reply:
yes, it use another CharArrayWriter, and in this class append() method does not call supper.append()  so CharArrayWriter is empty.